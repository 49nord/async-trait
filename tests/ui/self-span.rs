use async_trait::async_trait;

pub struct Struct {}

#[async_trait]
pub trait Trait {
    async fn method(self);
}

#[async_trait]
impl Trait for Struct {
    async fn method(self) {
        let _: () = self;
        let _: Self = Self;
    }
}

fn main() {}
