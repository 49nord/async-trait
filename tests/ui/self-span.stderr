error[E0423]: expected value, found struct `Struct`
  --> $DIR/self-span.rs:14:23
   |
3  | pub struct Struct {}
   | -------------------- `Struct` defined here
...
14 |         let _: Self = Self;
   |                       ^^^^ did you mean `Struct { /* fields */ }`?
   |
help: consider importing one of these items instead
   |
1  | use syn::Data::Struct;
   |
1  | use syn::Expr::Struct;
   |
1  | use syn::Item::Struct;
   |
1  | use syn::Pat::Struct;
   |
     and 1 other candidate

error[E0308]: mismatched types
  --> $DIR/self-span.rs:13:21
   |
13 |         let _: () = self;
   |                --   ^^^^ expected `()`, found struct `Struct`
   |                |
   |                expected due to this
